### DISCUSSION TITLE
[5 giugno 2023] Calcolabilità

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/automi-calcolabilita-complessita#11] [Calcolabilità]
![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/69389461/ae3d44c6-4796-45cb-9060-c18eecc3e55f)

**Comment by Exyss** [comment]
Per il primo punto, ripropongo la soluzione presente sui miei appunti:

![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/65043207/6333c150-34fc-478b-983e-c6380b6404fb)

----

La dimostrazione richiesta nel secondo punto è presente sul libro di testo

**Comment by aflaag** [comment]
È possibile dimostare il primo punto in maniera analoga alla soluzione di @Exyss , come segue:

$F$, su input $\textlangle M,w \textrangle$, computa come segue:
  - $F$ costruisce una macchina di Turing $M'$ che, su input $x$, computa come segue:
    - $M'$ è composta da $q_0, q_\mathrm{accept}, q_\mathrm{reject}$ e stati che servono per controllare se $x$ coincida o meno con $w$
    - se $x = w$, $M'$ accetta se $M(w)$ accetta
    - se $x \neq w$, $M'$ accetta

La dimostrazione è equivalente poiché se $M$ accetta $w$, allora $M'$ accetta tutte le stringhe, e dunque $q_\mathrm{reject}$ è il suo stato inutile, mentre se $M$ rifiuta $w$, allora $M'$ accetta ogni stringa diversa da $w$, eccetto $w$ stessa, dunque non ha stati inutili.

**Comment by CasuFrost** [comment]
![image](https://github.com/user-attachments/assets/4d834446-1799-46b2-a61a-f24f702ec97f)
![image](https://github.com/user-attachments/assets/95138264-43d4-4d2e-b68c-7686e04edeef)
![image](https://github.com/user-attachments/assets/4c9dd541-5739-4df2-88a1-b2b4c54a766d)
